metpy/io/upperair.py

```diff
@@ -167,15 +167,54 @@ def parse(fobj):
             fobj.readline()
 
         # Parse the actual data, only grabbing the columns for pressure, T/Td, and wind
-        cols = (0, 2, 3, 6, 7)
         unit_strs = ['degC' if u == 'C' else u
                      for u in fobj.readline().decode('ascii').split()]
 
-        # Skip 2 header lines -- 1 for '----' and 1 for partial line at 1000mb
-        p, t, td, direc, spd = np.genfromtxt(fobj, usecols=cols, skip_header=2, unpack=True)
-
-        return dict(p=(p, unit_strs[0]), t=(t, unit_strs[2]), td=(td, unit_strs[3]),
-                    wind=(direc, spd, unit_strs[7]))
+        # Parse to the actually beginning of data at or above ground.
+        fobj.readline()
```

Comment:
Do we still need this part? I think removing this would improve readability at the cost of having a point or two of missing data. This would also ensure that the "mandatory" levels are in the result.

At the very least, hard-coding the check for 11 items would seem to preclude levels near the surface where only some of the fields are missing...